bin/npg_heron_metadata_upload

```diff
@@ -167,9 +168,17 @@ foreach my $id_run (@id_run) {
 
     my %library_sample_ids;
     foreach my $flowcell (@flowcells) {
+      my $row4string = sub {
```

Comment:
Does this need to be an anonynmous sub? It's always called with the same flowcell argument and can just be a string:

```
my $msg = sprintf q[run %d, position %i, tag_index %i],
    $id_run, $frow->position, $frow->tag_index;

$library_name or croak "id_pool_lims is not defined for $msg";

```